core/src/main/java/org/jivesoftware/sparkimpl/certificates/CertificateModel.java

```diff
@@ -129,8 +169,227 @@ public CertificateModel(X509Certificate certificate) {
+					CRLDistributionPointsExtension = Res.getString("cert.is.critical") + critical + "\n";
+					CRLDistributionPointsExtension += point.toString();
+					extensions.put(oid, CRLDistributionPointsExtension);
+				} catch (IOException e) {
+					Log.warning("Couldn't extract CRL Distribution Points extension from certificate", e);
+				}
+
+			} else if (oid.equals("2.5.29.33")) {
+				ASN1OctetString oct = ASN1OctetString.getInstance(cert.getExtensionValue(oid));
+				policyMappingsExtension = Res.getString("cert.is.critical") + critical + "\n";
+				policyMappingsExtension += oct.toString();
+				extensions.put(oid, policyMappingsExtension);
+			} else if (oid.equals("2.5.29.35")) {
+				try {
+					AuthorityKeyIdentifier authorityKeyIdentifier = AuthorityKeyIdentifier
+							.fromExtensions(new JcaX509CertificateHolder(cert).getExtensions());
+					authorityKeyIdentifierExtension = Res.getString("cert.is.critical") + critical + "\n";
+					authorityKeyIdentifierExtension += Hex.toHexString(authorityKeyIdentifier.getKeyIdentifier());
+				} catch (CertificateEncodingException e) {
+					Log.warning("Couldn't extract authority key identifier extension", e);
+				}
+				extensions.put(oid, authorityKeyIdentifierExtension);
+			} else if (oid.equals("2.5.29.36")) {
+				try {
+					primitive = JcaX509ExtensionUtils.parseExtensionValue(cert.getExtensionValue(oid));
+					PolicyConstraints pc = PolicyConstraints.getInstance(primitive);
+					policyConstraintsExtension = Res.getString("cert.is.critical") + critical + "\n";
+					if (pc.getInhibitPolicyMapping() != null) {
+						policyConstraintsExtension += Res
+								.getString("cert.extension.policy.constraints.inhibit.policy.mapping") + ": "
+								+ pc.getInhibitPolicyMapping() + "\n";
+					}
+					if (pc.getRequireExplicitPolicyMapping() != null) {
+						policyConstraintsExtension += Res
+								.getString("cert.extension.policy.constraints.require.explicit.policy.mapping") + ": "
+								+ pc.getRequireExplicitPolicyMapping();
+					}
+					extensions.put(oid, policyConstraintsExtension);
+				} catch (IOException e) {
+					Log.warning("Couldn't extract policy constraints exception", e);
+				}
+
+			} else if (oid.equals("2.5.29.37")) {
+				try {
+					extendedKeyUsageExtension = Res.getString("cert.is.critical") + critical + "\n";
+					List<String> extKeyUsage = cert.getExtendedKeyUsage();
+					for (String use : extKeyUsage) {
+						extendedKeyUsageExtension += use + ": " + OIDTranslator.getDescription(use) + "\n";
+					}
+					extensions.put(oid, extendedKeyUsageExtension);
+				} catch (CertificateParsingException e) {
+					Log.warning("Couldn't parse extended key usage extension", e);
+				}
+			} else {
+				addToUnsupported(critical, oid);
+			}
+		} catch (NullPointerException e) {
+			Log.error("Couldn't extract " + oid + ": " + OIDTranslator.getDescription(oid) + "extension.", e);
+			addToUnsupported(critical, oid);
+		}
```

Comment:
This is getting a very, very long method, which will be hard to maintain. Can we improve on the code style here?